src/index.ts

```diff
@@ -0,0 +1,72 @@
+import { Request, Response } from 'express';
+import * as _ from 'lodash';
+import { version } from '../package.json';
+import { getCsvDataStream } from './csv';
+import { TransformsList } from 'adlib';
+import { ServiceError } from './csv/service-error';
+
+export = class OutputCsv2 {
+  static type = 'output';
+  static version = version;
+  static routes = [
+    {
+      path: '/csv',
+      methods: ['get'],
+      handler: 'serve',
+    },
+  ];
+
+  model: any;
+
+  public async serve(req: Request, res: Response) {
+    res.setHeader('Content-Type', 'text/csv');
+    try {
+      const csvTemplate = _.get(req, 'res.locals.csvTemplate') as Record<string, any>;
+      const csvTemplateTransforms = _.get(req, 'app.locals.csvTemplateTransforms') as TransformsList;
+      const csvFileName = _.get(req, 'app.locals.csvFileName', 'dataset');
```

Comment:
I think we may want to avoid this use of `dataset` here as well. Mainly because we use that word all over our ecosystem to mean very different things. What about a default name of `output` instead?

Reply:
I agree. I have changed it to `output`.